cmd/snap-confine/user-support.c

```diff
@@ -38,7 +38,7 @@ void setup_user_data(void)
 
 	debug("creating user data directory: %s", user_data);
 	if (sc_nonfatal_mkpath(user_data, 0755) < 0) {
-		if (errno == EROFS && !sc_startswith(user_data, "/home/")) {
+		if ((errno == EROFS || errno == EACCES) && !sc_startswith(user_data, "/home/")) {
```

Comment:
what is the situation when EACCES happen?

Reply:
It's described in the corresponding commit message: https://github.com/snapcore/snapd/pull/11367/commits/2d380b9863a22970cf4a9e17b5270ca946e7ceef